core/server/themes/index.js

```diff
@@ -22,8 +23,17 @@ module.exports = {
         // Just read the active theme for now
         return themeLoader
             .loadOneTheme(activeThemeName)
-            .then(function activeThemeHasLoaded() {
-                debug('Activating theme (method A on boot)', activeThemeName);
+            .then(function activeThemeHasLoaded(theme) {
+                // Validate
+                return validate
```

Comment:
> If we don't use gscan, then we have to re-implement the reading of templates, partials, etc outside of gscan. We already use gscan for uploads and activations, it makes no sense to me to implement the same logic again in Ghost and do the same work twice?

yeah this is true. I haven't seen that we use the result of the gscan validation.

> Perhaps there are ways to make gscan faster / more efficient?

yes. will look at gscan tomorrow (dependencies and requires).